### PR TITLE
New option to ignore the basenamespace

### PR DESCRIPTION
**Pull request by lauort**
Hi,

I modified the wsdl, create a new test to check when necessarily the user wants to ignore the base name space if it exist.

The problem was:

The code always uses the base name space of a node if it exists, in some cases this is wrong because the server can not accept this workaround and you need to ignore that base namespaces. The problem is localized in the line 1557 in the wsdl:

```java
if(childParameterTypeObject.$baseNameSpace 
{
    ns = childParameterTypeObject.$baseNameSpace + ':';
}
```

Solution: 

To use it:

```javascript
options: {
"ignoreBaseNameSpaces" : true/false (by default false)
}
```

I added a new option: "ignoreBaseNameSpaces" by default set "false", and modified the line, now it looks like:

```java
if(childParameterTypeObject.$baseNameSpace && !this.options.ignoreBaseNameSpaces) {
   ns = childParameterTypeObject.$baseNameSpace + ':';
}
```

Also I modified the "request-response-samples-test.js"

Because the call adds the "options" parameter in the end, that is incorrect because the call never use that var in the code. Now the call makes sense with the signature to call the "createClient"

Old call:

```java
 soap.createClient(wsdlPath, wsdlOptions, function(err, client){.......}, options);
```

New call:

```java
 soap.createClient(wsdlPath, wsdlOptions, function(err, client){.......});
```

I update also de Readme to add the new wsdl_options description.

Thats all.

Regards

### CODE REVIEW

**Comment by riaan53** [comment]
This works perfectly and solves my issue.

Thanks @lauort !

**Comment by lauort** [comment]
You are welcome @riaan53  :dancer:

**Comment by lauort** [comment]
Hi @herom 

Does it exist the possibility to merge this pull request in the master branch?? I am using this module in my project and it is very important.

Regards.

**Comment by riaan53** [comment]
+1

On Thursday, July 30, 2015, lauort <notifications@github.com> wrote:

> Hi @herom <https://github.com/herom>
>
> Does it exist the possibility to merge this pull request in the master
> branch?? I am using this module in my project and it is very important.
>
> Regards.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vpulim/node-soap/pull/689#issuecomment-126240594>.
>


-- 
Riaan Jacobs
riaan53@gmail.com

**Comment by droidenator** [comment]
This worked for me! Thank you so much. This issue has been causing me a lot of trouble and this patch fixed it.

**Comment by lauort** [comment]
Thanks @droidenator , but it is needed to merge with the master branch X_X!

File: test/request-response-samples-test.js [hunk 1]
```diff
@@ -119,7 +119,7 @@ function generateTest(name, methodName, wsdlPath, headerJSON, securityJSON, requ
           }
         }
         done();
-      }, options);
```
Comment (jsdevel) [inline test/request-response-samples-test.js]: please revert this

**Comment by jsdevel** [comment]
Looks pretty good to me, other than it needs to be squashed down to 1 commit.

**Comment by lauort** [comment]
Hi @jsdevel, 

I put again that var in test file and pushed it.

Regards!

**Comment by lauort** [comment]
Hi @jsdevel,

is there any problem to merge my changes?

Regards

**Comment by jsdevel** [comment]
@lauort almost there:

* Please squash your changes down to a single commit in this branch's history
* Please add an example in the README.

**Comment by lauort** [comment]
Hi @jsdevel 

I made a reset to the branch, and update the readme, added the option, I think is all right now.

Regards

**Comment by jsdevel** [comment]
I'm still seeing 10 commits.

**Comment by lauort** [comment]
Hi @jsdevel 

Now is only 1 commit i do not know why the git do not update my previously head reset.
Regards and sorry for all the drawbacks.

**Comment by jsdevel** [comment]
I'm still seeing multiple commits:

multiple commits

**Comment by lauort** [comment]
Can you esplain me what is the problem becaus if i reset the branch, all the changes will be lose... do you prefer i open a new pull request?

**Comment by lauort** [comment]
I am making 1 commit only and I will make the pull request

**Comment by lauort** [comment]
HI @jsdevel 

Now the branch only has 1 commit with all the changes.

New Test, add descriptions in the 2 readme files, and the modifications in the wsdl file.

Sorry for all but is the 1st time i tried to colaborate in a git project.

Regards and thanks for all.

**Comment by riaan53** [comment]
@lauort looks like github is complaining about conflicts still? "This branch has conflicts that must be resolved"

Cant wait until this is in :)

**Comment by lauort** [comment]
@riaan53 I think the merge problems need to be solved by  @jsdevel  if i need make some other changes i will :dancer:

**Comment by lauort** [comment]
Hi  @jsdevel , Do i need to make some rebase to solve the issues or you will fix it?

Regards

**Comment by jsdevel** [comment]
You'll need to rebase your branch onto the latest in master.  There are several ways to do this.  I'd likely do:

1. `git checkout master`
2. `git pull --rebase upstream master`
3. resolve conflicts in your editor of choice, `git add` them, then `git rebase --continue`
4. `git push -f origin master`

**Comment by lauort** [comment]
Hi @jsdevel i will make a new pull request because my system bring me mad... deleted my master branch. Sorry for all the problems

**Pull request closed without merging**